python/lsst/sims/photUtils/Sed.py

```diff
@@ -81,14 +81,253 @@
+
+            for file_name in file_list:
+                if file_name.endswith('.gz'):
+                    try:
+                        full_name = os.path.join(dir_name, file_name)
+                        data = numpy.genfromtxt(full_name, dtype=dtype)
+                        cache[full_name] = (data['wavelen'], data['flambda'])
+                    except:
+                        pass
+
+    with open(os.path.join(cache_dir, cache_name), "wb") as file_handle:
+        pickle.dump(cache, file_handle)
+
+    # record the specific sims_sed_library directory being cached so that
+    # a new cache will be generated if sims_sed_library gets updated
+    with open(os.path.join(cache_dir, "cache_version.txt"), "w") as file_handle:
+        file_handle.write("%s %s" % (sed_root, cache_name))
+
+    return cache
+
+
+def cache_LSST_seds():
+    """
+    Read all of the SEDs in sims_sed_library into a dict.  Pickle the dict
+    and store it in sims_photUtils/cacheDir/lsst_sed_cache.p for future use.
+
+    After the file has initially been created, the next time you run this script,
+    it will just use pickle to load the dict.
+
+    Once the dict is loaded, Sed.readSED_flambda() will be able to read any
+    LSST-shipped SED directly from memory, rather than using I/O to read it
+    from an ASCII file stored on disk.
+
+    Note: the dict of cached SEDs will take up about 5GB on disk.  Once loaded,
+    the cache will take up about 1.5GB of memory.  The cache takes about 14 minutes
+    to generate and about 51 seconds to load on a 2014 Mac Book Pro.
+    """
+
+    global _global_lsst_sed_cache
+    try:
+        sed_cache_dir = os.path.join(getPackageDir('sims_photUtils'), 'cacheDir')
+        sed_cache_name = os.path.join('lsst_sed_cache.p')
+        sed_dir = getPackageDir('sims_sed_library')
+
+    except:
+        raise
+        print("You did not install sims_photUtils with the full LSST simulations "
+              "stack. You cannot generate and load the cache of LSST SEDs")
+        return
+
+    must_generate = False
+    if not os.path.exists(os.path.join(sed_cache_dir, sed_cache_name)):
+        must_generate = True
+    if not os.path.exists(os.path.join(sed_cache_dir, "cache_version.txt")):
+        must_generate = True
+    else:
+        with open(os.path.join(sed_cache_dir, "cache_version.txt"), "r") as input_file:
+            lines = input_file.readlines()
+            if len(lines) != 1:
+                must_generate = True
+            else:
+                info = lines[0].split()
+                if len(info) != 2:
+                     must_generate = True
+                elif info[0] != sed_dir:
+                    must_generate = True
+                elif info[1] != sed_cache_name:
+                    must_generate = True
+
+    if must_generate:
+        print "creating cache of LSST SEDs"
+        cache = _generate_sed_cache(sed_cache_dir, sed_cache_name)
+        _global_lsst_sed_cache = cache
```

Comment:
Maybe also a print statement to say the cache is complete and where it's stored?